creator-node/src/snapbackSM/peerSetManager.js

```diff
@@ -0,0 +1,221 @@
+const axios = require('axios')
+
+const { logger } = require('../logging')
+
+// TODO: config var
+const PEER_HEALTH_CHECK_REQUEST_TIMEOUT = 2000
+
+class PeerSetManager {
+  constructor ({ discoveryProviderEndpoint, creatorNodeEndpoint }) {
+    this.discoveryProviderEndpoint = discoveryProviderEndpoint
+    this.creatorNodeEndpoint = creatorNodeEndpoint
+  }
+
+  log (msg) {
+    logger.info(`SnapbackSM:::PeerSetManager: ${msg}`)
+  }
+
+  logError (msg) {
+    logger.error(`SnapbackSM:::PeerSetManager ERROR: ${msg}`)
+  }
+
+  /**
+   * Retrieves node users, paritions it into a slice, computes the peer set of the node users,
+   * and performs a health check on the peer set.
+   * @param {Object[]} nodeUsers array of objects of schema { primary, secondary1, secondary2, user_id, wallet }
+   * @returns the unhealthy peers in a Set
+   *
+   * @note consider returning healthy set?
+   */
+  async getUnhealthyPeers (nodeUsers) {
+    // Compute content node peerset from nodeUsers (all nodes that are in a shared replica set with this node)
+    let peerSet = this.computeContentNodePeerSet(nodeUsers)
+
+    /**
+     * Determine health for every peer & build list of unhealthy peers
+     * TODO: change from sequential to chunked parallel
+     */
+    const unhealthyPeers = new Set()
+
+    for await (const peer of peerSet) {
+      try {
+        // TODO: consider using the response
+        await this.determinePeerHealth(peer)
+      } catch (e) {
+        unhealthyPeers.add(peer)
+        this.logError(`getUnhealthyPeers() peer=${peer} is unhealthy: ${e.toString()}`)
+      }
+    }
+
+    return unhealthyPeers
+  }
+
+  /**
+   * Retrieve list of all users which have this node as replica (primary or secondary) from discovery node
+   * Or retrieve primary users only if connected to old discprov
+   *
+   * Also handles backwards compatibility of getAllNodeUsers() and getNodePrimaryUsers()
+   * This only works if both functions have a consistent return format
+   */
+  async getNodeUsers () {
+    let fetchUsersSuccess = false
```

Comment:
this is the same as the snapbackSM.js `getNodeUsers()` method, with some slight modification on:
- handling the error on both node user fetching